pepper-apis/dss-core/src/main/java/org/broadinstitute/ddp/model/activity/definition/TabularBlockDef.java

```diff
@@ -0,0 +1,54 @@
+package org.broadinstitute.ddp.model.activity.definition;
+
+import lombok.Value;
+import org.broadinstitute.ddp.model.activity.definition.question.QuestionDef;
+import org.broadinstitute.ddp.model.activity.definition.tabular.TabularHeaderDef;
+import org.broadinstitute.ddp.model.activity.definition.tabular.TabularRowDef;
+import org.broadinstitute.ddp.model.activity.types.BlockType;
+
+import javax.validation.Valid;
+import javax.validation.constraints.NotEmpty;
+import javax.validation.constraints.NotNull;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+import java.util.stream.Stream;
+
+@Value
+public class TabularBlockDef extends FormBlockDef {
+    int columnsCount;
+
+    @NotEmpty
+    List<@Valid @NotNull TabularHeaderDef> headers = new ArrayList<>();
```

Comment:
Did you intend to use the `javax.validation` annotations here instead of the Lombok annotations?

Reply:
Yep. This is similar to what we have in other DEF classes. Should be used by the hibernate validator.

Reply:
My understanding is we are only using the Hibernate Validator to check POJOs deserialized by GSON from the endpoints and, IIRC, against some Templates loaded from study builder- and we do not have a more general use of the library. Historically, the patterns in the `Def` classes (and elsewhere that is not JSON-adjacent) were intended to act as a hint to static analysis tools and for developers, and are effectively NOPs.

Reply:
Am I correct in assuming that your suggestion is to remove those annotations?

Reply:
I was thinking they are in use when we read the Def file from the study definition.

Reply:
In this case, let's remove the `@NotEmpty`, and `@Valid`, and use Lombok's `@NotNull`. If the property is guaranteed to both be non-`null` AND not empty, please add that as part of the description in a Javadoc comment for the property.

Reply:
Then it is better to remove all the annotations since it makes no sense. Valid annotation suppose to say hibernate to validate nested objects. In case of removal them, NotEmpty is also not required since headers and rows are always being initialized. I'll remove all of them.

Reply:
Most likely you were referring to `NonNull` annotation, right?